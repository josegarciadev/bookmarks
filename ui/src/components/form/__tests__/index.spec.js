import { shallow } from "enzyme";

import { BookmarkForm } from "../index";

describe("BookmarkForm", () => {
  let onHandleChangeMock, onHandleSaveMock;

  beforeEach(() => {
    onHandleChangeMock = jest.fn();
    onHandleSaveMock = jest.fn();
  });
  let props = {
    onHandleChange: onHandleChangeMock,
    onHandleSave: onHandleSaveMock,
    form: {
      title: "",
      link: "",
    },
  };

  it("should match snapshot", () => {
    const rendered = shallow(<BookmarkForm {...props} />);
    expect(rendered).toMatchSnapshot();
  });

  it("should match Onchange Event", () => {
    const rendered = shallow(
      <BookmarkForm
        {...props}
        onHandleSave={onHandleSaveMock}
        onHandleChange={onHandleChangeMock}
      />
    );

    rendered
      .find("input")
      .at(0)
      .simulate("change", {
        target: { name: "title", value: "Test react app" },
      });
    rendered
      .find("input")
      .at(1)
      .simulate("change", {
        target: { name: "link", value: "http://link.com" },
      });

    rendered.find("form").simulate("submit");
    expect(onHandleChangeMock).toHaveBeenCalled();
    expect(onHandleSaveMock).toHaveBeenCalled();
    expect(rendered).toMatchSnapshot();
  });

  it("should match Onchange empty", () => {
    props.form.title = "test";
    const rendered = shallow(
      <BookmarkForm
        {...props}
        onHandleSave={onHandleSaveMock}
        onHandleChange={onHandleChangeMock}
      />
    );

    expect(onHandleSaveMock).toHaveBeenCalled();
    expect(rendered).toMatchSnapshot();
  });
});
