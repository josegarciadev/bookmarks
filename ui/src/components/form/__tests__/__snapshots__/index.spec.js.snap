// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkForm should match Onchange Event 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label
        htmlFor="title"
      >
        Title: 
      </label>
      <input
        className="form-input-text"
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label
        className="FormLabel"
        htmlFor="Link"
      >
        Link:
      </label>
      <input
        className="form-input-text"
        id="Link"
        name="link"
        onChange={[Function]}
        type="url"
        value=""
      />
    </div>
    <input
      className="buttom-add"
      disabled={true}
      type="submit"
      value="Add"
    />
  </form>
</Fragment>
`;

exports[`BookmarkForm should match Onchange empty 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label
        htmlFor="title"
      >
        Title: 
      </label>
      <input
        className="form-input-text"
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label
        className="FormLabel"
        htmlFor="Link"
      >
        Link:
      </label>
      <input
        className="form-input-text"
        id="Link"
        name="link"
        onChange={[Function]}
        type="url"
        value=""
      />
    </div>
    <input
      className="buttom-add"
      disabled={true}
      type="submit"
      value="Add"
    />
  </form>
</Fragment>
`;

exports[`BookmarkForm should match snapshot 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label
        htmlFor="title"
      >
        Title: 
      </label>
      <input
        className="form-input-text"
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label
        className="FormLabel"
        htmlFor="Link"
      >
        Link:
      </label>
      <input
        className="form-input-text"
        id="Link"
        name="link"
        onChange={[Function]}
        type="url"
        value=""
      />
    </div>
    <input
      className="buttom-add"
      disabled={true}
      type="submit"
      value="Add"
    />
  </form>
</Fragment>
`;
